Nakama/INTransport.cs

```diff
@@ -16,46 +40,46 @@ internal interface INTransport
             Action<byte[]> successAction,
             Action<Exception> errorAction);
 
-        void Connect(string uri, bool noDelay);
-        void ConnectAsync(string uri, bool noDelay, Action<bool> callback);
+        void Connect(string uri, byte[] token);
+        void ConnectAsync(string uri, byte[] token, Action<bool> callback);
         void Close();
         void CloseAsync(Action callback);
-        void Send(byte[] data);
-        void SendAsync(byte[] data, Action<bool> completed);
+        void Send(byte[] data, bool reliable);
+        void SendAsync(byte[] data, bool reliable, Action<bool> completed);
 
-        event EventHandler<WebSocketCloseEventArgs> OnClose;
-        event EventHandler<WebSocketErrorEventArgs> OnError;
-        event EventHandler<WebSocketMessageEventArgs> OnMessage;
+        event EventHandler<SocketCloseEventArgs> OnClose;
```

Comment:
If we can make this change mostly internal lets get rid of the event handler usage and just use `System.Action<T>`. The event handler stuff in C# doesn't work well enough yet in Unity.

Reply:
But if it would introduce other complexity we can leave it for now.